test/jdk/jdk/internal/util/ArchTest.java

```diff
@@ -0,0 +1,132 @@
+     */
+    @Test
+    public void nameVsCurrent() {
+        String osArch = System.getProperty("os.arch").toLowerCase(Locale.ROOT);
+        System.out.printf("System property os.arch: \"%s\", Architecture.current(): \"%s\"%n",
+                osArch, Architecture.current());
+        Architecture arch = switch (osArch) {
+            case "x86_64" -> X64;
+            case "x86" -> X86;
+            case "i386" -> X86;
+            case "amd64" -> X64;  // Is alias for X86_64
+            case "aarch64" -> AARCH64;
+            case "riscv64" -> RISCV64;  // unverified
+            case "s390x", "s390" -> S390;  // unverified
+            case "ppc64le" -> PPC64;  // unverified
+            default -> OTHER;
+        };
+        assertEquals(Architecture.current(), arch, "mismatch in Architecture.current vs " + osArch);
+    }
+
+    /**
+     * Test various Architecture enum values vs boolean isXXX() methods.
+     * @return a stream of arguments for parameterized test
+     */
+    private static Stream<Arguments> archParams() {
+        return Stream.of(
+                Arguments.of(X64, Architecture.isX64()),
+                Arguments.of(X86, Architecture.isX86()),
+                Arguments.of(AARCH64, Architecture.isAARCH64()),
+                Arguments.of(RISCV64, Architecture.isRISCV64()),
+                Arguments.of(S390, Architecture.isS390()),
+                Arguments.of(PPC64, Architecture.isPPC64())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("archParams")
+    public void isArch(Architecture arch, boolean isArch) {
+        Architecture current = Architecture.current();
+        assertEquals(arch == current, isArch,
+                "Mismatch " + arch + " == " + current + " vs is" + arch);
+    }
+
+    /**
+     * Test that Architecture.is64bit() matches Architecture.current().
+     */
+    @Test
+    public void is64BitVsCurrent() {
+        Architecture current = Architecture.current();
+        boolean expected64Bit = switch (current) {
+            case OTHER -> Architecture.is64bit();   // Always ok, expected value is unknown
+            case X64 -> true;
+            case X86 -> false;
+            case AARCH64 -> true;
+            case RISCV64 -> true;
+            case S390 -> true;
+            case PPC64 -> true;
+        };
+        assertEquals(Architecture.is64bit(), expected64Bit, "mismatch in is64bit");
+    }
+
+    /**
+     * Test that Architecture.isLittleEndian() matches Architecture.current().
+     */
+    @Test
+    public void isLittleEndianVsCurrent() {
+        Architecture current = Architecture.current();
+        boolean expectedEndian = switch (current) {
+            case OTHER -> Architecture.isLittleEndian();   // Always ok, expected value is unknown
+            case X64 -> true;
+            case X86 -> true;
+            case AARCH64 -> true;
+            case RISCV64 -> true;
+            case S390 -> false;
+            case PPC64 -> true;
```

Comment:
This is not always true. The PPC64 architecture supports both endianness versions. AIX and legacy linux is Big Endian while recent linux is little Endian (determined by platform name "os.arch = ppc64le" instead of "os.arch = ppc64"). Querying the endianness is also possible, of course.

Reply:
```suggestion
            case PPC64 -> !OperatingSystem.isAix() && Architecture.isLittleEndian();
```

Reply:
Looks correct, but makes the test pointless for any linux on PPC64.